pytube/contrib/search.py

```diff
@@ -0,0 +1,198 @@
+"""Module for interacting with YouTube search."""
+# Native python imports
+import logging
+
+# Local imports
+from pytube import YouTube
+from pytube.innertube import InnerTube
+
+
+logger = logging.getLogger(__name__)
+
+
+class Search:
+    def __init__(self, query):
+        self.query = query
```

Comment:
maybe call it rawQuery?

Reply:
since it is input by user

Reply:
Yeah, I missed the docstring for __init__, that's my bad. I'll fix that. I don't think raw_query makes much sense since we aren't transforming the user's query behind the scenes, and it only makes other lines longer unnecessarily.

Reply:
makes sense - if you could just add a comment :-) 

my thinking is it would differentiate between user-initiated searches and system-initiated searches (as part of a different feature), but perhaps I've just working too much in midtiers recently

Reply:
We definitely shouldn't be initiating searches ourselves because of network overhead. I also can't imagine a use case where we need to query YouTube's search functionality on behalf of a user where the user wouldn't query it themselves. If we need to provide functionality for that, we can just create new `Search` objects and return those to the user, similar to how I provide `YouTube` objects to the user from search results